integration-tests/helpers/index.js

```diff
@@ -271,15 +271,18 @@ async function createSandbox (dependencies = [], isGitRepo = false,
   }
   const folder = path.join(os.tmpdir(), id().toString())
   const out = path.join(folder, `dd-trace-${version}.tgz`)
-  const allDependencies = [`file:${out}`].concat(cappedDependencies)
 
   await fs.mkdir(folder)
+  await execHelper('command -v bun >/dev/null 2>&1 || npm i -g bun@1.2.23', { env: restOfEnv })
   const preferOfflineFlag = process.env.OFFLINE === '1' || process.env.OFFLINE === 'true' ? ' --prefer-offline' : ''
-  const addCommand = `yarn add ${allDependencies.join(' ')} --ignore-engines${preferOfflineFlag}`
   const addOptions = { cwd: folder, env: restOfEnv }
-  await execHelper(`npm pack --silent --pack-destination ${folder}`, { env: restOfEnv }) // TODO: cache this
+  await execHelper(`bun pm pack --silent --destination ${folder}`, { env: restOfEnv }) // TODO: cache this
```

Comment:
If we're switching packers in here, we should either switch to whatever we're using at publish time (yarn I think?) or switch our publish job to use Bun.

Overall, I think this is not the best solution specifically for ESM tests, since they really have no business being integration tests in the first place. Let's undo that, first.